packages/api/src/middlewares/body.ts

```diff
@@ -0,0 +1,96 @@
+import type { PieceContext } from '@sapphire/framework';
+import { URLSearchParams } from 'url';
+import { createGunzip, createInflate } from 'zlib';
+import type { ApiRequest } from '../lib/structures/api/ApiRequest';
+import type { ApiResponse } from '../lib/structures/api/ApiResponse';
+import { HttpCodes } from '../lib/structures/http/HttpCodes';
+import { Middleware } from '../lib/structures/Middleware';
+import type { Route } from '../lib/structures/Route';
+import { MimeTypes } from '../lib/utils/MimeTypes';
+
+export class PluginMiddleware extends Middleware {
+	public constructor(context: PieceContext) {
+		super(context, { priority: 20 });
+	}
+
+	public run(request: ApiRequest, response: ApiResponse, route: Route) {
+		const contentType = request.headers['content-type'];
+		if (typeof contentType !== 'string') return null;
+
+		const lengthString = request.headers['content-length'];
+		const length = lengthString ? Number(lengthString) : Number.MAX_SAFE_INTEGER;
+		if (length > route.maximumBodyLength) {
+			response.status(HttpCodes.PayloadTooLarge).json({ error: 'Exceeded maximum content length.' });
+			return null;
+		}
+
+		const index = contentType.indexOf(';');
+		const type = index === -1 ? contentType : contentType.slice(0, index);
+
+		switch (type) {
+			case MimeTypes.ApplicationJson:
+				return this.applicationJson(request, response);
+			case MimeTypes.ApplicationFormUrlEncoded:
+				return this.applicationFormUrlEncoded(request, response);
+			case MimeTypes.TextPlain:
+				return this.textPlain(request);
+			default:
```

Comment:
where's my raw buffer

Reply:
Needs more mime types, also, allowing unhandled binary/raw buffer can result on a security issue as clients will be able to overload the server by sending very large binary buffers.

Reply:
Except you already have a size check, albeit miniscule, and you can already crash it with JSON or other forms